pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1195,12 +1234,24 @@ func (c *Controller) isPxImageBeingUpdated(toUpdate *corev1.StorageCluster) bool
 		(toUpdate.Spec.Version == "" || newVersion != toUpdate.Status.Version)
 }
 
-func getZoneMap(nodeList *v1.NodeList) map[string]uint64 {
+func getZoneMap(nodeList *v1.NodeList, filterLabelKey string, filterLabelValue string) map[string]uint64 {
 	cloudProviderName := getCloudProviderName(nodeList)
 	cloudProvider := cloudprovider.New(cloudProviderName)
 	zoneMap := map[string]uint64{}
 	for _, node := range nodeList.Items {
 		if zone, err := cloudProvider.GetZone(&node); err == nil {
+			if len(filterLabelKey) > 0 {
+				value, ok := node.Labels[filterLabelKey]
```

Comment:
In px we look for `portworx.io/node-type` key suffix and not exact match. I don't remember the reason, but just to be consistent, we should do the same here.

Reply:
Thanks for pointing that. I hadn't noticed it.
@adityadani do you know why we have a suffix match and not an exact match?

Reply:
I also don't remember the reason. In hindsight I think this should have been an exact match.